processor/src/main/java/org/mapstruct/ap/internal/model/assignment/HasCheckWrapper.java

```diff
@@ -0,0 +1,62 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.model.assignment;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.mapstruct.ap.internal.model.common.Type;
+
+/**
+ * Wraps the assignment in a target setter if CheckHas is set to true.
+ *
+ * @author Sean Huang
+ */
+public class HasCheckWrapper extends AssignmentWrapper {
```

Comment:
Could you add to the JavaDoc the snippet of code it generates (or, if feasible, do it via the `toString()`) method; Esp. to make the difference to `UpdateHasCheckWrapper` more apparent.